src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs

```diff
@@ -197,17 +197,28 @@ public ProjectContext Build()
                 libraries.Add(new LibraryKey(mainProject.Identity.Name), mainProject);
             }
 
+            LibraryRange? platformDependency = null;
+            if (mainProject != null)
+            {
+                platformDependency = mainProject.Dependencies.FirstOrDefault(d => d.Type.Equals(LibraryDependencyType.Platform));
+            }
+
             LockFileTarget target = null;
             if (lockFileLookup != null)
             {
-                target = SelectTarget(LockFile);
+                target = SelectTarget(LockFile, platformDependency == null);
                 if (target != null)
                 {
                     var nugetPackageResolver = new PackageDependencyProvider(PackagesDirectory, frameworkReferenceResolver);
                     var msbuildProjectResolver = new MSBuildDependencyProvider(Project, ProjectResolver);
                     ScanLibraries(target, lockFileLookup, libraries, msbuildProjectResolver, nugetPackageResolver, projectResolver);
                 }
             }
+            LibraryDescription platformLibrary = null;
+            if(platformDependency != null)
+            {
+                platformLibrary = libraries[new LibraryKey(platformLibrary.Identity.Name, platformLibrary.Identity.Type)];
```

Comment:
Did you mean to use platformDependency.Identity here? Otherwise this will be a null pointer.